src/ConnectKafka.cpp

```diff
@@ -65,38 +90,45 @@ KafkaMessageMetadataStruct ConnectKafka::consumeFromOffset() {
   KafkaMessageMetadataStruct DataToReturn;
 
   auto KafkaMsg = std::unique_ptr<Message>(Consumer->consume(1000));
-  switch (KafkaMsg->err()) {
-  case RdKafka::ERR_NO_ERROR:
-    // Real message
-    if (KafkaMsg->len() > 0) {
-      std::string Payload(static_cast<const char *>(KafkaMsg->payload()),
-                          static_cast<int>(KafkaMsg->len()));
-      DataToReturn.Payload = Payload;
-      DataToReturn.Partition = KafkaMsg->partition();
-      DataToReturn.Offset = KafkaMsg->offset();
-      DataToReturn.Timestamp = KafkaMsg->timestamp().timestamp;
+  try {
```

Comment:
I don't think this `try-catch` block is necessary, you are already handling the errors from `rdkafka` there shoudn't be an exception thrown.